File: src/JourneyJournal.Data/Entities/Expense.cs
using System.ComponentModel.DataAnnotations;
using JourneyJournal.Data.Enums;

namespace JourneyJournal.Data.Entities;

/// <summary>
/// Tracks additional trip expenses with categorization.
/// </summary>
public class Expense
{
    /// <summary>
    /// Primary key.
    /// </summary>
    public int ExpenseId { get; set; }

    /// <summary>
    /// Foreign key to Trip.
    /// </summary>
    [Required]
    public int TripId { get; set; }

    /// <summary>
    /// Expense description.
    /// </summary>
    [Required]
    [MaxLength(200)]
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// Expense category (Transportation, Food, etc.).
    /// </summary>
    [Required]
    public ExpenseCategory Category { get; set; }

    /// <summary>
    /// Expense amount.
    /// </summary>
    [Required]
    public decimal Amount { get; set; }

    /// <summary>
    /// Date of expense.
    /// </summary>
    [Required]
    public DateTime ExpenseDate { get; set; }

    /// <summary>
    /// Payment method (Cash, Credit Card).
    /// </summary>
    [Required]
    public PaymentMethod PaymentMethod { get; set; }

    /// <summary>
    /// Record creation timestamp (UTC).
    /// </summary>
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Record last update timestamp (UTC).
    /// </summary>
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties

    /// <summary>
    /// Parent trip.
    /// </summary>
    public Trip Trip { get; set; } = null!;
}

File: src/JourneyJournal.Web/src/types/expense.ts
export interface Expense {
  expenseId: number;
  tripId: number;
  description: string;
  category: ExpenseCategory;
  amount: number;
  expenseDate: string;
  paymentMethod: PaymentMethod;
  notes?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateExpenseRequest {
  description: string;
  category: ExpenseCategory;
  amount: number;
  expenseDate: string;
  paymentMethod: PaymentMethod;
  notes?: string;
}

export interface UpdateExpenseRequest {
  description: string;
  category: ExpenseCategory;
  amount: number;
  expenseDate: string;
  paymentMethod: PaymentMethod;
  notes?: string;
}

export const ExpenseCategory = {
  Transportation: 1,
  Restaurant: 2,
  Food: 3,
  Entertainment: 4,
  Shopping: 5,
  Fee: 6,
  Living: 7,
  Other: 8
} as const;

export type ExpenseCategory = typeof ExpenseCategory[keyof typeof ExpenseCategory];

export const PaymentMethod = {
  Cash: 1,
  CreditCard: 2
} as const;

export type PaymentMethod = typeof PaymentMethod[keyof typeof PaymentMethod];
